presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -550,7 +550,11 @@ public void rollback()
     public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle handle, Constraint constraint)
     {
         IcebergTableHandle table = (IcebergTableHandle) handle;
-        TupleDomain<IcebergColumnHandle> newDomain = convertTupleDomainTypes(constraint.getSummary().transform(IcebergColumnHandle.class::cast));
+        TupleDomain<IcebergColumnHandle> newDomain = convertTupleDomainTypes(
+                constraint.getSummary()
+                        .transform(IcebergColumnHandle.class::cast)
+                        .simplify())
```

Comment:
Once iceberg supports `IN` predicate, it is not necessary to `simplify()`. Can you add TODO comment and explain why simplify() is necessary.